### PR TITLE
Add a callback to stop training by max_steps.

### PR DESCRIPTION
**Pull request by workingloong**
After the master receives a completed training task, we can add up the training steps (the count of mini-batches) in the task. If the total training steps > max_steps, we will set `callback_list.model.stop_training=True` and then clear `TaskDispatcher.todo` list. However, the master must wait that the tasks in`TaskDispatcher.doing` are completed and the training steps may be a little bigger than `max_steps`.

### CODE REVIEW

**Comment by skydoorkai** [comment]
What if the training is loaded from a checkpoint? `_completed_steps` will be 0. Probably assign this value from model_version if loaded from a checkpoint.

**Comment by workingloong** [comment]
> What if the training is loaded from a checkpoint? `_completed_steps` will be 0. Probably assign this value from model_version if loaded from a checkpoint.

Good suggestion. We can get the model_version from the checkpoint file.

File: elasticdl/python/common/save_utils.py [hunk 1]
```diff
@@ -259,3 +259,11 @@ def restore_params_from_checkpoint(checkpoint_dir, shard_index, shard_num):
         parameters.embedding_params.update(embedding_tables)
         parameters.version = version
         return parameters
+
+    @staticmethod
+    def get_version_from_checkpoint(checkpoint_dir):
+        variable_shard_files = os.listdir(checkpoint_dir)
+        shard_file_path = os.path.join(checkpoint_dir, variable_shard_files[0])
```
Comment (brightcoder01) [inline elasticdl/python/common/save_utils.py:270]: From this code, we only load the first file in variable_shard_files, what about other files?

File: elasticdl/python/master/master.py [hunk 2]
```diff
@@ -127,6 +144,18 @@ def __init__(self, args):
         self._should_stop = False
         self._exit_code = 0
 
+    def _set_completed_steps_by_checkpoint(self, checkpoint_dir_for_init):
```
Comment (brightcoder01) [inline elasticdl/python/master/master.py:150]: When will we call this method?

File: elasticdl/python/common/save_utils.py (hunk 1 above)
Comment (workingloong) [reply elasticdl/python/common/save_utils.py:270]: Each shard file has the version and the versions in shard files are the same. So, we only need to load one file to get the version.

File: elasticdl/python/master/master.py (hunk 2 above)
Comment (workingloong) [reply elasticdl/python/master/master.py:150]: We should call this method using `args.checkpoint_dir_for_init` after initializing callbacks. I have added the code to call it in the next commit.

File: elasticdl/python/master/master.py [hunk 3]
```diff
@@ -127,6 +147,18 @@ def __init__(self, args):
...
+        if not checkpoint_dir_for_init:
+            return
+
+        CheckpointSaver.check_checkpoint_valid(checkpoint_dir_for_init)
```
Comment (brightcoder01) [inline elasticdl/python/master/master.py]: If check_checkpoint_valid returns false, we need return directly instead of executing the following logic

Comment (workingloong) [reply elasticdl/python/master/master.py]: Yes, it may be better to raise an error if checkpoint dir is not None and invalid.

**Review by brightcoder01 — APPROVED** [review]

File: elasticdl/python/master/master.py (hunk 3 above)
Comment (skydoorkai) [inline elasticdl/python/master/master.py]: If not valid, return

Comment (workingloong) [reply elasticdl/python/master/master.py]: Yes, I have added the code to raise a `ValueError` if invalid.

**Pull request merged**